pkg/applicationserver/applicationserver.go

```diff
@@ -567,16 +573,14 @@ func (as *ApplicationServer) handleJoinAccept(ctx context.Context, ids ttnpb.End
 				StartedAt: time.Now().UTC(),
 			}
 			mask = append(mask, "pending_session")
-			if !joinAccept.PendingSession && len(joinAccept.InvalidatedDownlinks) > 0 {
-				// The Network Server reset the downlink queue as the new security session invalidated it.
-				// The invalidated downlink queue is passed as part of the join-accept and the Application Server should recalculate it.
-				// This changes the LastAFCntDown in the session, so it should be run as part of the transaction.
+			if len(joinAccept.InvalidatedDownlinks) > 0 {
+				// The Network Server does not reset the downlink queues as the new security session is established.
```

Comment:
Is this true @rvolosatovs ?

Reply:
> @rvolosatovs so is NS now discarding the current queue on join-accept, or not? The items in there are invalid for sure, so I would expect so, hence the original comment here.

No, NS discards the queue as part of session switch when the uplink confirming the new session is received. It does send the "current" queue to AS as part of join-accept.